include/ppx/geometry.h

```diff
@@ -157,6 +167,11 @@ class Geometry
         char*       GetData() { return DataPtr(mData); }
         const char* GetData() const { return DataPtr(mData); }
         uint32_t    GetElementCount() const;
+        uint32_t    GetDataSize() const;
+
+        // Append() will automatically adjust mOffset, but it may need to be
+        // manually adjusted if Overwrite() is used
+        void SetOffset(uint32_t offset) { mOffset = offset; }
```

Comment:
it is risky to expose this as public since people can call this and then call Append() which is not the desired behavior i assume? can this be part of 'Overwrite()`?

Reply:
I'm not sure it's possible since I think `Overwrite()` can be used in different scenarios:
1. Writing new data to buffer that has already been resized at the start (mOffset needs to be adjusted) (in my example, the part where the single sphere is repeated into the larger buffer)
2. Overwriting existing data in buffer at a point earlier than mOffset (mOffset does not need to be adjusted) (in my example, the part where unique position data is overwriting the standard position data)

I understand this is risky and that's why the old flow of using Append instead of Overwrite and not setting a final size is intended as the less risky but slower option.

Reply:
I think the problem is that the user needs to dig into the implementation to understand how to use this function which is something we should avoid because this could make the class difficult to use and error-prone. 
Wondering if it is only related to `Overwrite`, can it be somehow a input arg of `Overwrite`?